plugins/py3embed/py3embed.cpp

```diff
@@ -1205,7 +1241,11 @@ PyObject* ECLDatasetIterator_iternext(PyObject *self)
     ECLDatasetIterator *p = (ECLDatasetIterator *)self;
     if (p->val)
     {
-        roxiemem::OwnedConstRoxieRow nextRow = p->val->ungroupedNextRow();
+        roxiemem::OwnedConstRoxieRow nextRow;
+        {
+            GILUnblock b;
+            nextRow.setown(p->val->ungroupedNextRow());
+        }
         if (!nextRow)
         {
             p->val->stop();
```

Comment:
I think the stop should also occur with the unblock.  It is unlikely, but possible that it may cause some python to be executed.

Reply:
Fixed